DNN Platform/Library/Entities/Modules/Settings/ISettingsRepository.cs

```diff
@@ -7,8 +7,36 @@ namespace DotNetNuke.Entities.Modules.Settings
     public interface ISettingsRepository<T>
         where T : class
     {
-        T GetSettings(ModuleInfo moduleContext);
-
+        /// <summary>
+        /// Retrieve module settings. This can optionally include TabModule settings, Portal settings and
+        /// host settings as well. Note the result is cached.
+        /// </summary>
+        /// <param name="moduleContext">Your module's context.</param>
+        /// <returns>Your settings class.</returns>
+        T GetSettings(ModuleInfo moduleContext);
+
+        /// <summary>
+        /// Retrieve portal/host settings. This will ignore Module and TabModule settings.
+        /// Note the result is cached.
+        /// </summary>
+        /// <param name="portalId">The portal ID for which to retrieve the settings.</param>
+        /// <returns>Your settings class.</returns>
+        T GetSettings(int portalId);

```

Comment:
Given the differences in behaviors here, I think it might be better for these two new methods to be added with different names, as we are blending/mixing Module and Portal settings, in a manner that is only clear if you dive into the comments of the particular method. Additionally, the differentiation between Portal & Host settings is hard to understand here.  If a behavior is host settings with a -1 value, we should document this.

Given we also have ModuleSettings, TabSettings, and TabModuleSettings "GetSettings" is just too ambiguous in my opinion.

Reply:
This is part of the settings repository, which retrieves a "settings object".  That object's properties will be marked with attributes to determine which settings they're retrieved from.  So, typically, most of the settings will be module settings and used within the context of a module, but potentially the same pattern can be used in instances where you know that there aren't module settings, or you only care about properties that you know are portal settings.  But `GetSettings` does retrieve all relevant settings with the information it's been given.

Reply:
Ok, I'm fine, just seems confusing to mix portal/module/host here with the same API even with it being generic